org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacTypeBinding.java

```diff
@@ -609,14 +630,15 @@ public String getName() {
 			return builder.toString();
 		}
 		StringBuilder builder = new StringBuilder(this.typeSymbol.getSimpleName().toString());
-		ITypeBinding[] types = this.getUncheckedTypeArguments(this.type, this.typeSymbol);
-
-		if (types != null && types.length > 0) {
-			builder.append("<");
-			for (var typeArgument : types) {
-				builder.append(typeArgument.getName());
+		if( this.specificInstance) {
```

Comment:
I think it would be more reliable to replace this with
```suggestion
		if (isParameterized()) {
```
which is better capturing JDT expecations and would allow us to only focus better on more narrowly scoped methods.